client/blocks/post-likes/index.jsx

```diff
@@ -3,81 +3,133 @@
 /**
  * External dependencies
  */
-
 import React from 'react';
+import classnames from 'classnames';
 import { connect } from 'react-redux';
-import { flowRight } from 'lodash';
 import { localize } from 'i18n-calypso';
 
 /**
  * Internal dependencies
  */
 import Gravatar from 'components/gravatar';
+import Gridicon from 'gridicons';
 import QueryPostLikes from 'components/data/query-post-likes';
 import { isRequestingPostLikes, getPostLikes, countPostLikes } from 'state/selectors';
 import { recordGoogleEvent } from 'state/analytics/actions';
 
-export const PostLikes = props => {
-	const { countLikes, isRequesting, likes, postId, postType, siteId, translate } = props;
-	const trackLikeClick = () => props.recordGoogleEvent( 'Post Likes', 'Clicked on Gravatar' );
-	const getLikeUrl = like => {
+class PostLikes extends React.PureComponent {
+	static defaultProps = {
+		postType: 'post',
+		showDisplayNames: false,
+	};
+
+	getLikeUrl = like => {
 		return like.URL ? like.URL : `https://gravatar.com/${ like.login }`;
 	};
 
-	let noLikesLabel;
+	trackLikeClick = () => {
+		this.props.recordGoogleEvent( 'Post Likes', 'Clicked on Gravatar' );
+	};
+
+	renderLike = like => {
+		const { showDisplayNames } = this.props;
+
+		return (
+			<a
+				key={ like.ID }
+				href={ this.getLikeUrl( like ) }
+				rel="noopener noreferrer"
+				target="_blank"
+				className="post-likes__item"
+				onClick={ this.trackLikeClick }
+			>
+				<Gravatar user={ like } alt={ like.login } title={ like.login } size={ 24 } />
+				{ showDisplayNames && (
+					<span className="post-likes__display-name">
+						{ like.name }
+						<Gridicon className="post-likes__external-link-icon" icon="external" size={ 12 } />
+					</span>
+				) }
+			</a>
+		);
+	};
+
+	renderExtraCount() {
+		const { likes, likeCount, showDisplayNames, translate } = this.props;
+
+		if ( ! likes || likeCount <= likes.length ) {
+			return null;
+		}
+
+		const extraCount = likeCount - likes.length;
+
+		let message;
+		if ( showDisplayNames ) {
+			message = translate( '+ %(extraCount)d more', '+ %(extraCount)d more', {
```

Comment:
Minor nit: since we're introducing a new string here anyways, should we take the opportunity to use `numberFormat`?

Reply:
Sounds good to me.  I hadn't used this function before, but I changed the devdocs popover example to a post with more likes, followed the pattern in [this code](https://github.com/Automattic/wp-calypso/blob/0f69e72/client/my-sites/plugins/plugin-ratings/index.jsx#L89-L106), and it's working well.